spec/rails_translation_manager/locale_checker/incompatible_plurals_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe IncompatiblePlurals do
+  context 'when there are missing plurals' do
+    let(:all_locales) do
+      [
+        {
+          locale: :en,
+          keys: %w[browse.some_key.other browse.some_other_key.one]
+        }
+      ]
+    end
+
+    it 'returns the missing plural forms' do
+      expect(described_class.new(all_locales).report)
+        .to eq(
+          <<~OUTPUT.chomp
+            \e[31m[ERROR]\e[0m Incompatible plural forms, for:
+
+            - 'en', with parent 'browse.some_key'. Expected: [:one, :other], actual: [:other]
+
+            - 'en', with parent 'browse.some_other_key'. Expected: [:one, :other], actual: [:one]
+          OUTPUT
+        )
+    end
+  end
+
+  context "when there aren't any missing plurals" do
+    let(:all_locales) do
+      [
+        {
+          locale: :en,
+          keys: %w[browse.some_key.other browse.some_key.one]
+        }
+      ]
+    end
+
+    it 'returns nil' do
+      expect(described_class.new(all_locales).report)
+        .to be_nil
+    end
+  end
+
+  context "when plural forms aren't available for the locales" do
+    let(:all_locales) do
+      [
+        {
+          locale: :en,
+          keys: %w[browse.some_key.other]
```

Comment:
I think it might be helpful to answer your question by first explaining where the pluralization rules come from. All pluralization rules, E.g for English: "one" and "other" are [set from a directory](https://github.com/svenfuchs/rails-i18n/tree/master/rails/pluralization) in `rails-i18n` which is included in Rails Translation Manager. This list has rules for quite a few languages, but not all the ones we support. Thats where the [plurals.rb](https://github.com/alphagov/rails_translation_manager/pull/12/files#diff-ae6999ab8352836786563d07f5e23f47c46440176c54f50e91f01aec9096a190R3) comes into play - it fills the gap for the languages that don't have pluralization rules natively in `rails-i18n`. These lists both get added to `I18n`'s load path during runtime and provide I18n the ability to work out which plural to use for each language, depending on the number of items in question.

The solution implemented here (and in govuk-app-config) is to use a mix of `rails-i18n` and a custom list to support all our languages. I did some thinking / a mini spike into just using a static list that we maintain, which includes all the plural rules for all the languages we support. However, this would mean we would be solely responsible for maintaining the list. As the plural rules in `rails-i18n` are updated fairly regularly (last language added was in May), then it seems like a good idea to use their list and hopefully one day, as it gets updated, we could then remove our custom one.

To get back to you question concerning the difference between the two tests. The first one is testing whether the plural rules are correct for the keys and locale given (English). As English uses "one" and "other" then these two keys must be present for each plural group, for example `key.one, key.other`. In the first test there is  `browse.some_key.other` and `browse.some_other_key.one`, as these two keys are distinct then for each there is a missing plural: `browse.some_key.one` and `browse.some_other_key.other`. 

The test which this comment is attached: `"when plural forms aren't available for the locales"` is concerned with when neither `rails-i18n` or `plurals.rb` contains a plural form (rule) for the language. This causes the test to fail due to not being able to work out the correct plurals. This probably will occur when a new language is added. The next action from this error message would be to add the plural rule for the language into `plurals.rb`. 

Sorry if I've waffled / not answered your question properly, do let me know if you have any more questions. I can talk it though via a call if needed. It is quite confusing!